avocado/utils/cpu.py

```diff
@@ -90,42 +90,62 @@ def cpu_has_flags(flags):
     return True
 
 
-def get_cpu_vendor_name():
+def get_version():
+    """
+    Get cpu version
+
+    :returns: string cpu version of given machine
+    E:g:- 'i5-5300U' for intel and 'POWER9' for ibm machines
+    incase for unknown/unhandled machines return empty string
+    """
+    version_pattern = {'x86_64': rb'\s([\S,\d]+)\sCPU',
+                       'i386': rb'\s([\S,\d]+)\sCPU',
+                       'powerpc': rb'revision\s+:\s+(\S+)',
+                       's390': rb'.*machine\s=\s(\d+)'
+                       }
+    cpu_info = _get_info()
+    arch = get_arch()
+    try:
+        version_pattern[arch]
+    except KeyError as Err:
+        logging.warning("No pattern string for arch: %s\n Error: %s", arch, Err)
+        return ""
+    for line in cpu_info:
+        version_out = re.findall(version_pattern[arch], line)
+        if version_out:
+            break
+    return version_out[0].decode('utf-8') if version_out[0] else ""
+
+
+def get_vendor():
     """
     Get the current cpu vendor name
 
-    :returns: string 'intel' or 'amd' or 'power7' depending on the
+    :returns: string 'intel' or 'amd' or 'ibm' depending on the
          current CPU architecture.
     :rtype: `string`
     """
     vendors_map = {
         'intel': (b"GenuineIntel", ),
         'amd': (b"AMD", ),
-        'power7': (b"POWER7", ),
-        'power8': (b"POWER8", ),
-        'power9': (b"POWER9", )
+        'ibm': (rb"POWER\d", rb"IBM/S390", ),
     }
 
-    cpu_info = _get_cpu_info()
+    cpu_info = _get_info()
     for vendor, identifiers in vendors_map.items():
         for identifier in identifiers:
             if _list_matches(cpu_info, identifier):
                 return vendor
     return None
 
 
-def get_cpu_arch():
+def get_arch():
```

Comment:
All of those need to contain backward compatibility methods with warning for transition period...